### ISSUE TITLE
Can't search with both category and group filters

### ISSUE THREAD

**Issue by msiemens** [akeneo/pim-community-dev#11148]
When searching for products, and setting both category and groups filters, no results are returned. The filters are configured as displayed in this screenshot:

![Screenshot 2019-11-29 16 05 36](https://user-images.githubusercontent.com/1873922/69877323-2e6be600-12c2-11ea-8e1f-e2b03ac0f9aa.png)

When removing either the groups filter or setting the categories filter to „All products“, results are shown as expected.

---

I've tried to dig into this a little and it seems like when searching for categories, `ProductAndProductModelSearchAggregator` configures a `mustNot` filter for the `categories_of_ancestors` attribute in Elasticsearch, excluding all products that have parent models that are assigned to the selected category. That way only product models are returned. But then the groups filter removes all product model search results as groups are defined on the product level, not on the product model level.

In other words, it seems like the categories filter only matches product models. But setting the group filter will only match on products. Thus there are no search results.

---

System information report: 
[akeneo-pim-system-info_2019-11-29_15_03.txt](https://github.com/akeneo/pim-community-dev/files/3905715/akeneo-pim-system-info_2019-11-29_15_03.txt)

**Comment by msiemens** [comment]
@valerie-schneider or anyone else from the support team: is it possible to get feedback on this bug?

**Comment by valerie-schneider** [comment]
Hello Markus, 

I do reproduce this, and your guessing is right. By default, the grid displays only simple products and product models, but not product variants. If the products in your group are all variant products, it can explain you don't see any. 
The workaround I could suggest you is to force the grid to display only simple products, by filtering (for example) on _SKU (identifier) does not contain "akeneo"_ (or another word). This way, only products (i.e. simple products and variant products) will be displayed on the grid, as product models don't have any SKU. 

Have a good day

Valérie

**Comment by msiemens** [comment]
Thanks for the workaround! Could you forward this bug report to the dev team so bug can be fixed properly?
